torch/utils/data/graph_settings.py

```diff
@@ -45,13 +47,29 @@ def apply_sharding(datapipe: DataPipe, num_of_instances: int, instance_id: int)
     return datapipe
 
 
-def apply_shuffle_settings(datapipe: DataPipe, shuffle: Optional[bool]) -> DataPipe:
+def _is_shuffle_datapipe(datapipe: DataPipe) -> bool:
+    if not hasattr(datapipe, "set_shuffle") or not hasattr(datapipe, "set_seed"):
+        return False
+    if not inspect.ismethod(datapipe.set_shuffle) or not inspect.ismethod(datapipe.set_seed):
+        return False
+    return True
+
+
+def apply_shuffle_settings(datapipe: DataPipe, shuffle: Optional[bool] = None) -> DataPipe:
+    r"""
+    Traverse the graph of ``DataPipes`` to find and set shuffle attribute
+    to each `DataPipe` that has an API of ``set_shuffle``.
```

Comment:
If I'm reading `_is_shuffle_datapipe` correctly, it requires both ``set_shuffle`` and ``set_seed`` to exist right?
```suggestion
    to each `DataPipe` that has an API of ``set_shuffle`` and ``set_seed``.
```